UITests/ContentBlockerSettingsTest.swift

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import EarlGrey
+import Deferred
+@testable import Client
+
+class ContentBlockerSettingsTests: KIFTestCase {
+    var tab: Tab!
+    var delegate: AppDelegate!
+    var testDone: XCTestExpectation!
+
+    // A convenience function for static or instance usage.
+    func tapInTable(_ label: String) {
+        ContentBlockerSettingsTests.tapInTable(label)
+    }
+
+    class func tapInTable(_ label: String) {
+        EarlGrey.select(elementWithMatcher:grey_accessibilityLabel(label)).using(searchAction: grey_scrollInDirection(GREYDirection.down, 600), onElementWithMatcher: grey_kindOfClass(UITableView.self)).perform(grey_tap())
+    }
+
+    override class func setUp() {
+        super.setUp()
+        // Called once before all tests are run
+        BrowserUtils.dismissFirstRunUI()
+        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("Menu")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("SettingsMenuItem")).perform(grey_tap())
+        tapInTable("Tracking Protection")
+    }
+
+    override class func tearDown() {
+        let delegate = UIApplication.shared.delegate as! AppDelegate
+        delegate.tabManager.removeAll()
+    }
+
+    override func setUp() {
+        testDone = expectation(description: "wait for test")
+        delegate = UIApplication.shared.delegate as! AppDelegate
+        tab = delegate.tabManager.addTab()
+    }
+
+    @available(iOS 11, *)
+    func testStrict() {
+        let blocker = tab.contentBlocker as! ContentBlockerHelper
+        tapInTable("Always On")
+        tapInTable("Strict")
+        blocker.addActiveRulesToTab().uponQueue(.main) { result in
```

Comment:
In tests you dont need to use closures like this. You can use `blocker.addActiveRulesToTab().value` and this will block until the result is returned. 😄

Reply:
.value doesn't work in these test cases, it hangs and doesn't continue, is this expected to work? I can look again where it was sticking.

Reply:
So far `all(deferreds) ` within addActiveRulesToTab() never executes using .value in the test case. I'll try it outside of test and see if .value works there.

Reply:
It works fine outside of test.

Reply:
```
 func doThing() -> Deferred<Void> {
            let result = Deferred<Void>()

            let rules = [1,2,3]
            let deferreds: [Deferred<Void>] = rules.map { value in
                let result = Deferred<Void>()
                DispatchQueue.main.async {
                    result.fill(())
                }
                return result
            }

            all(deferreds).uponQueue(.main) { list in
                result.fill(()) // never called in .value case
            }

            return result
        }
```

Reply:
Are you sure .value works as you expect? It locks up in that above block of code.